pkg/liqo-controller-manager/namespaceMap-controller/update_namespaceoffloading_status.go

```diff
@@ -0,0 +1,150 @@
+package namespacemapctrl
+
+import (
+	"context"
+	"fmt"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/klog"
+
+	offv1alpha1 "github.com/liqotech/liqo/apis/offloading/v1alpha1"
+	liqoconst "github.com/liqotech/liqo/pkg/consts"
+	liqoutils "github.com/liqotech/liqo/pkg/utils"
+)
+
+// If all requested remote namespaces have been created, it is necessary to set the global status of Offloading,
+// otherwise the Offloading is still in progress.
+func (r *NamespaceMapReconciler) checkNamespaceOffloadingStatus(namespaceOffloading *offv1alpha1.NamespaceOffloading) error {
+	virtualNodes := &corev1.NodeList{}
+	if err := r.List(context.TODO(), virtualNodes,
+		client.MatchingLabels{liqoconst.TypeLabel: liqoconst.TypeNode}); err != nil {
+		klog.Error(err, " --> Unable to List all virtual nodes")
+		return err
+	}
+
+	if len(virtualNodes.Items) == 0 {
+		err := fmt.Errorf(" No VirtualNode at the moment")
```

Comment:
AFAIK This should not be an error condition. It is legitimate not to have an active peering in a given moment

Reply:
If the NamespaceMap involved in the reconcile process exists, at least the associated virtualNode must exist, I have added a check of NotFound condition at the beginning of the reconcile function.